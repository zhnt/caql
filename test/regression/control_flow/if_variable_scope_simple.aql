// Simplified variable scope in if statements (nested block without if condition)
let global_var = "global"

if 1 == 1 {
    let block_var = "block"
    print("Inside if: " + global_var)
    print("Inside if: " + block_var)
    
    {
        let nested_var = "nested"
        print("Nested block: " + global_var)
        print("Nested block: " + block_var)
        print("Nested block: " + nested_var)
        
        // Modify outer variables
        global_var = "modified_global"
        block_var = "modified_block"
    }
    
    print("After nested: " + global_var)
    print("After nested: " + block_var)
} else {
    print("This should not execute")
}

print("Final global: " + global_var)
print("Variable scope test done")
