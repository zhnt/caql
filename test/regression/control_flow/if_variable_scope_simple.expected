Inside if: global
Inside if: block
Nested block: global
Nested block: block
Nested block: nested
After nested: modified_global
After nested: modified_block
Final global: modified_global
Variable scope test done
